src/Microsoft.Data.SqlClient/src/Microsoft/Data/Common/AdapterUtil.Windows.cs

```diff
@@ -40,10 +43,12 @@ internal static object LocalMachineRegistryValue(string subkey, string queryvalu
                 ADP.TraceExceptionWithoutRethrow(e);
                 return null;
             }
+#if !NET6_0_OR_GREATER
             finally
             {
-                RegistryPermission.RevertAssert();
+                CodeAccessPermission.RevertAssert();
```

Comment:
do we need a `#else` block for `RegistryPermission.RevertAssert()`?

Reply:
I do not think so. The only thing that this finally does is reverting Assert. which is obsolete in net6.

Reply:
CAS methods like assertion and reversion are no-ops in netcore. So you can execute them without worrying. They are obsolete and will flag up as compiler warnings though so if we want a warning free build it might be a good idea to wrap them in an #ifdef NETFX

Reply:
> it might be a good idea to wrap them in an #ifdef NETFX

I thought about them, but NetStandard builds, and tests were worrying me.